### PR TITLE
Add note on diff's n differing from pandas

### PR DESCRIPTION
**Pull request by max-sixty**
This surprised me so I thought worth calling out. 

Ref https://github.com/pydata/xarray/pull/3800

### CODE REVIEW

**Comment by dcherian** [comment]
I guess pandas now calls this `periods`? https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.diff.html

**Comment by max-sixty** [comment]
Oh good spot @dcherian . I'll add that to the note

**Comment by max-sixty** [comment]
Updated

File: xarray/core/dataarray.py [hunk 1]
```diff
@@ -2692,6 +2692,12 @@ def diff(self, dim: Hashable, n: int = 1, label: Hashable = "upper") -> "DataArr
         difference : same type as caller
             The n-th order finite difference of this object.
 
+        .. note::
+
+            `n` matches numpy's behavior and is different from pandas' similarly
```
Comment (dcherian) [inline xarray/core/dataarray.py]: I think "similarly located" is confusing.

How about 

```
The first argument `n` matches numpy's behavior and is different from pandas' first argument named `periods`.
```

Comment (max-sixty) [reply xarray/core/dataarray.py]: Yes much better

**Comment by dcherian** [comment]
Thanks @max-sixty

**Comment by max-sixty** [comment]
Ended up being more from you than me @dcherian :)

**Pull request merged**